### PR TITLE
chore: Address linting issues in apps/

### PR DESCRIPTION
**Pull request by bznein**
## Description

Addresses linting issues (mostly unchecked type assertions and redundant imports) in apps.

In a specific case (inside modules/apps/callbacks/testing/simapp/app.go) I had to add `panic` as we don't return errors from there and we are not within a testing environment.  But I'm open to more complex refactoring if we want to more gracefully handle these cases.

ref: #6086 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [x] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [x] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

## Summary by CodeRabbit

- **Refactor**
	- Enhanced type assertion checks and error handling in various test suites to improve reliability.
	- Simplified import statements across several modules for better code readability.
- **Bug Fixes**
	- Adjusted logic for service skipping based on annotations, ensuring correct behavior in specific scenarios.

### CODE REVIEW

File: modules/apps/callbacks/testing/simapp/app.go [hunk 1]
```diff
@@ -511,7 +511,11 @@ func NewSimApp(
 	var transferStack porttypes.IBCModule
 	transferStack = transfer.NewIBCModule(app.TransferKeeper)
 	transferStack = ibccallbacks.NewIBCMiddleware(transferStack, app.IBCFeeKeeper, app.MockContractKeeper, maxCallbackGas)
-	transferICS4Wrapper := transferStack.(porttypes.ICS4Wrapper)
+	transferICS4Wrapper, ok := transferStack.(porttypes.ICS4Wrapper)
+	if !ok {
+		panic("Can't convert transferStack to ICS4Wrapper")
```
Comment (crodriguezvega) [inline modules/apps/callbacks/testing/simapp/app.go:516]: I think you could do here also something like...

```suggestion
		panic(fmt.Errorf("cannot convert %T to %T", transferStack, porttypes.ICS4Wrapper))
```

File: modules/apps/callbacks/testing/simapp/app.go [hunk 2]
```diff
@@ -526,10 +530,16 @@ func NewSimApp(
 	// initialize ICA module with mock module as the authentication module on the controller side
 	var icaControllerStack porttypes.IBCModule
 	icaControllerStack = ibcmock.NewIBCModule(&mockModule, ibcmock.NewIBCApp("", scopedICAMockKeeper))
-	app.ICAAuthModule = icaControllerStack.(ibcmock.IBCModule)
+	app.ICAAuthModule, ok = icaControllerStack.(ibcmock.IBCModule)
+	if !ok {
+		panic("Can't convert icaControllerStack to IBCModule")
```
Comment (crodriguezvega) [inline modules/apps/callbacks/testing/simapp/app.go:535]: ditto

File: modules/apps/callbacks/testing/simapp/app.go [hunk 3]
```diff
@@ -526,10 +530,16 @@ func NewSimApp(
...
+	}
 	icaControllerStack = icacontroller.NewIBCMiddleware(icaControllerStack, app.ICAControllerKeeper)
 	icaControllerStack = ibccallbacks.NewIBCMiddleware(icaControllerStack, app.IBCFeeKeeper, app.MockContractKeeper, maxCallbackGas)
-	icaICS4Wrapper := icaControllerStack.(porttypes.ICS4Wrapper)
+	icaICS4Wrapper, ok := icaControllerStack.(porttypes.ICS4Wrapper)
+	if !ok {
+		panic("Can't convert icaControllerStack to ICS4Wrapper")
```
Comment (crodriguezvega) [inline modules/apps/callbacks/testing/simapp/app.go:541]: ditto

**Review by crodriguezvega — APPROVED** [review]

File: modules/apps/callbacks/testing/simapp/app.go (hunk 1 above)
Comment (bznein) [reply modules/apps/callbacks/testing/simapp/app.go]: Done, but as in all similar occurrences, I had to use an extra variable as passing the type directly (or `&porttypes.ICS4Wrapper{}` won't work. If there's a way to achieve this that I missed, please let me know! :)

File: modules/apps/27-interchain-accounts/host/keeper/keeper.go [hunk 4]
```diff
@@ -288,8 +288,14 @@ func newModuleQuerySafeAllowList() []string {
 			sd := fd.Services().Get(i)
 
 			// Skip services that are annotated with the "cosmos.msg.v1.service" option.
-			if ext := proto.GetExtension(sd.Options(), msgv1.E_Service); ext != nil && ext.(bool) {
-				continue
+			if ext := proto.GetExtension(sd.Options(), msgv1.E_Service); ext != nil {
+				val, ok := ext.(bool)
+				if !ok {
+					panic("can't convert extension to boolean")
```
[the remark this answers is not in the document]
Comment (bznein) [reply modules/apps/27-interchain-accounts/host/keeper/keeper.go:294]: Done

**Review by chatton — APPROVED** [review]
LGTM

**Review by chatton — APPROVED** [review]
LGTM

**Pull request merged**